kedro/contrib/io/feather/feather_local.py

```diff
@@ -0,0 +1,130 @@
+# NONINFRINGEMENT. IN NO EVENT WILL THE LICENSOR OR OTHER CONTRIBUTORS
+# BE LIABLE FOR ANY CLAIM, DAMAGES, OR OTHER LIABILITY, WHETHER IN AN
+# ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF, OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# The QuantumBlack Visual Analytics Limited ("QuantumBlack") name and logo
+# (either separately or in combination, "QuantumBlack Trademarks") are
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+#     or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""``FeatherLocalDataSet`` loads and saves data to a local feather format file
+(https://github.com/wesm/feather). The
+underlying functionality is supported by pandas, so it supports all operations the pandas supports.
+"""
+from pathlib import Path
+from typing import Any, Dict
+
+import pandas as pd
+
+from kedro.io.core import AbstractVersionedDataSet, DataSetError, Version
+
+
+class FeatherLocalDataSet(AbstractVersionedDataSet):
+    """``FeatherLocalDataSet`` loads and saves data to a local feather file. The
+    underlying functionality is supported by pandas, so it supports all
+    allowed pandas options for loading and saving csv files.
+
+    Example:
+    ::
+
+        >>> from kedro.contrib.io.feather import FeatherLocalDataSet
+        >>> import pandas as pd
+        >>>
+        >>> data = pd.DataFrame({'col1': [1, 2], 'col2': [4, 5],
+        >>>                      'col3': [5, 6]})
+        >>>
+        >>> data_set = FeatherLocalDataSet(filepath="test.feather",
+        >>>                                load_args=None,
+        >>>                                save_args={"index": False})
+        >>> data_set.save(data)
+        >>> reloaded = data_set.load()
+        >>>
+        >>> assert data.equals(reloaded)
+
+    """
+
+    def __init__(
+            self,
+            filepath: str,
+            load_args: Dict[str, Any] = None,
+            save_args: Dict[str, Any] = None,
+            version: Version = None,
+    ) -> None:
+        """Creates a new instance of ``FeatherLocalDataSet`` pointing to a concrete
+        filepath.
+
+        Args:
+            filepath: path to a feather file.
+            load_args: feather options for loading feather files.
+                Here you can find all available arguments:
+                https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_feather.html#pandas.read_feather
+                All defaults are preserved.
+            save_args: None
```

Comment:
Yes, It would be better to remove these unused, i will do that. 

I was in two minds to build it using the base feather implementation or pandas(so there was confusion). Pandas doesn't need any args.